packages/react/runtime/src/utils.ts

```diff
@@ -1,14 +1,33 @@
 // Copyright 2024 The Lynx Authors. All rights reserved.
 // Licensed under the Apache License Version 2.0 that can be found in the
 // LICENSE file in the root directory of this source tree.
+
+import { getCurrentVNode, getOwnerStack } from './debug/component-stack.js';
+
 export function isDirectOrDeepEqual(a: any, b: any): boolean {
   if (a === b) {
     return true;
   }
-  if (
-    typeof a == 'object' && a !== null && typeof b == 'object' && b !== null && JSON.stringify(a) === JSON.stringify(b)
-  ) {
-    return true;
+  try {
+    if (
+      typeof a == 'object' && a !== null && typeof b == 'object' && b !== null
+      && JSON.stringify(a) === JSON.stringify(b)
+    ) {
+      return true;
+    }
+  } catch (error) {
+    if (__DEV__ && (error as Error).message.includes('circular')) {
```

Comment:
> https://github.com/FormidableLabs/react-fast-compare/blob/6f7d8afe02e4480c32f5af16f571367cccd47abc/index.js#L127

Should we check this like `react-fast-compare` does?

Reply:
I've tested on both PrimJS, V8 and JSC and updated the error message